includes/class-wc-payments.php

```diff
@@ -1398,4 +1404,18 @@ public static function enqueue_dev_runtime_scripts() {
 			wp_enqueue_script( 'WCPAY_RUNTIME', plugins_url( 'dist/runtime.js', WCPAY_PLUGIN_FILE ), [], self::get_file_version( 'dist/runtime.js' ), true );
 		}
 	}
+
+	/**
+	 * Display payment request button separator.
+	 */
+	public static function display_express_checkout_button_separator_html() {
+		$should_hide = WC_Payments_Features::is_payment_request_enabled() && ! WC_Payments_Features::is_woopay_express_checkout_enabled() && ! WC_Payments_Features::is_link_enabled();
```

Comment:
I assume this flag is for us to avoid the situation when the separator is rendered both in this method and in `WC_Payments_Payment_Request_Button_Handler::display_payment_request_button_separator_html`. 

I think after we remove all the redundant callback hooks in `WC_Payments_Payment_Request_Button_Handler` I mentioned in another comment and if we render the separator for the Payment Request button on the _Pay for order_ page, we'll eventually be able to remove this flag. Please let me know if I missed anything or if there is any other need to have this flag? Thanks!

Reply:
>I assume this flag is for us to avoid the situation when the separator is rendered both in this method and in WC_Payments_Payment_Request_Button_Handler::display_payment_request_button_separator_html

This `$should_hide` check is actually used to hide the separator when only payment request is enabled. The default for payment request buttons is to be hidden with `display: none;`. Then some frontend JS will check if Apple Pay or Google Pay is available in the browser and subsequently show the button and separator.

Reply:
Many thanks for the explanation, Mike! 👍  This one is interesting. 

----------------------------------------

Is my understanding correct that the expected UI for the Payment Request button, being the only enabled Express Payment method available, is

<p align="center">
<img width=60% height=60% alt="image" src="https://user-images.githubusercontent.com/22319444/213148866-fd32aedc-7d4b-474e-97d9-40962866395c.png">
</p>

instead of
<p align="center">
<img width=60% height=60% alt="image" src="https://user-images.githubusercontent.com/22319444/213149011-f814fb27-36ce-4c37-a7f2-cde261f21cc3.png">
</p>
?

----------------------------------------

Another thing is that I cannot see the difference between the two cases: `display: none` is provided to the element's styles or not. It seems to me that the browser (I'm using Chrome) removes `display: none`. In effect, regardless of whether the `$should_hide` flag is `true` or `false`, the HTML element is always rendered the same. Are you experiencing the same, and should we find a solution to this?

Reply:
Thanks Timur! Regarding the expected UI when only Payment Request is enabled, we should expect to still the separator, such as in your second screenshot:

<p align="center">
<img width=60% height=60% alt="image" src="https://user-images.githubusercontent.com/22319444/213148866-fd32aedc-7d4b-474e-97d9-40962866395c.png">
</p>

This is what I'm experiencing now. Are you seeing no separator in this case?

>Another thing is that I cannot see the difference between the two cases: display: none is provided to the element's styles or not. It seems to me that the browser (I'm using Chrome) removes display: none. In effect, regardless of whether the $should_hide flag is true or false, the HTML element is always rendered the same. Are you experiencing the same, and should we find a solution to this?

It's a little confusing indeed. So whenever Payment Request is the only express payment method, `$should_hide` will be true and `display:none;` will be present in the HTML. This ensures that the Payment Request button and separator are hidden by default and displayed **only** when the browser has Apple Pay or Google Pay enabled. So, in Chrome, with Google Pay configured with a payment method, `display:none` is [removed by this .show() method](https://github.com/Automattic/woocommerce-payments/blob/f592aafc4bb27110c9718b01482146e2f9dbce4e/client/payment-request/index.js#L452) after the [HTML elements are collected above](https://github.com/Automattic/woocommerce-payments/blob/f592aafc4bb27110c9718b01482146e2f9dbce4e/client/payment-request/index.js#L443). 

To see this in action, enable only Payment Request express checkout methods and test using a different profile in Chrome that doesn't have a Google Pay card attached. You should see something like this: 

![image](https://user-images.githubusercontent.com/3473953/213199470-7eacd29b-16bb-4d9c-adec-a047f2d4e9c8.png)

Let me know if you have anything above is confusing or if you're seeing something different on your end.

Reply:
Wow, this was super tricky - I had no idea of the separate JS running in parallel, which actually adds the separator once again, and I wouldn't figure this out without you. 🥇 Thanks, I'm resolving the thread.